src/sortStudents.ts

```diff
@@ -1,16 +1,52 @@
-
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades',
+}
+
+export type SortOrder = 'asc' | 'desc';
+
+function findAverage(grades: number[]): number {
+  return grades.reduce((sum, grade) => sum + grade, 0) / grades.length;
 }
 
-// create SortOrder type
-export type SortOrder;
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+) : Student[] {
+  const copy = [...students];
+
+  copy.sort((a, b) => {
+    switch (sortBy) {
+      case SortType.Name:
+      case SortType.Surname:
+        return (order === 'asc' ? 1 : -1) * a[sortBy].localeCompare(b[sortBy]);
+
+      case SortType.Age:
+      case SortType.Married:
+        return (order === 'asc' ? 1 : -1) * (+a[sortBy] - +b[sortBy]);
+
+      case SortType.AverageGrade:
+        return (
+          (order === 'asc' ? 1 : -1)
+          * (findAverage(a.grades) - findAverage(b.grades))
```

Comment:
```suggestion
        return ((order === 'asc' ? 1 : -1)
          * (findAverage(a.grades) - findAverage(b.grades))
```
You can create a variable for `(order === 'asc' ? 1 : -1)` in order not to duplicate yourself